File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> list;
        int i=0, j=1;
        while(i<target.size() && j<=n){
            list.push_back("Push");
            if(j == target[i]) i++;       
            else list.push_back("Pop");
            j++;
        }
        return list;
    }
};

File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        List<String> list = new ArrayList<>();
        int i=0, j=1;
        while(i<target.length && j<=n){
            list.add("Push");
            if(j == target[i]) i++;       
            else list.add("Pop");
            j++;
        }
        return list;
    }
}